materials/QueryAPI.py

```diff
@@ -0,0 +1,54 @@
+import requests
+
+class MissingKeyError(Exception):
+    def __init__(self, value):
+        self.value = value
+    def __str__(self):
+        return repr(self.value)
+
+class WolframAPI(object):
+    def __init__(self, key = None):
+        """
+        Object initializing WolframAlpha Short Answers API. Raises an exception if a developer key is not given.
+        Queries to the API are passed via the .query() classmethod.
+        
+        App ID Keys specific to the Short Answers API may be requested via https://developer.wolframalpha.com
+        
+        Args:
+            key (str): WolframAlpha App ID
+        """
+        self.key = key
+        
+        if self.key is None:
+            ermsg = 'API key required for property lookup. Please generate a key at https://developer.wolframalpha.com'
+            raise MissingKeyError(ermsg)
```

Comment:
You probably can just raise a `ValueError` here, if `key is None` that's an error in the value